core/src/main/kotlin/com/alecstrong/sql/psi/core/postgresql/PostgreSql.bnf

```diff
@@ -122,13 +122,23 @@ delete_stmt_limited ::= [ {with_clause} ] DELETE FROM {qualified_table_name} [ W
 
 insert_stmt ::= [ {with_clause} ]
     INSERT INTO [ {database_name} '.' ] {table_name} [ AS {table_alias} ]
-    [ '(' {column_name} ( ',' {column_name} ) * ')' ]
-    {insert_stmt_values} [ ON CONFLICT DO NOTHING ] [ returning_clause ] {
+    [ '(' {column_name} ( ',' {column_name} ) * ')' ] {insert_stmt_values}
+    [ ON CONFLICT ( [conflict_target] DO NOTHING | conflict_target conflict_update ) ]
+    [ returning_clause ] {
   extends = "com.alecstrong.sql.psi.core.psi.impl.SqlInsertStmtImpl"
   implements = "com.alecstrong.sql.psi.core.psi.SqlInsertStmt"
   override = true
 }
 
+
+conflict_target ::= '(' {column_name} ( ',' {column_name} ) * ')'
+
+conflict_update ::= DO UPDATE SET conflict_assign ( ',' conflict_assign * )
+
+conflict_assign ::= {column_name} '=' conflict_value
+
+conflict_value ::= {expr | DEFAULT}
```

Comment:
```suggestion
conflict_value ::= <<expr ‘-1’>> | DEFAULT
```

expressions are weird in overriding grammars, probably worth have a test where you set a column to a integer or something to make sure we test that case

Reply:
I saw that in the MySQL bnf and didn't understand it. Checking the GrammerKit doco, I now know  <<>> is for application of a meta-rule. Fixing it and adding the test for UPDATE SET with an expression.